### PR TITLE
chore(): suppress -Wunused-but-set-variable warns

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

**Comment by dundargoc** [comment]
Frankly, I'd rather we remove the assertions than introduce yet another goofy macro. I'll leave

File: src/nvim/macros.h [hunk 1]
```diff
@@ -226,4 +226,10 @@
 
 #define EMPTY_POS(a) ((a).lnum == 0 && (a).col == 0 && (a).coladd == 0)
 
+// Mark unused variables with _UNUSED, so that we can suppress warnings from
+// gcc's -Wunused-value and -Wunused-but-set-variable.
+// Prefix with _ to not clash with xdiff/xinclude.h which defines UNUSED for use
+// with function arguments.
+#define _UNUSED(x) ((void)(x))
```
Comment (theHamsta) [inline src/nvim/macros.h:233]: wouldn't a macro for assert be better that expands to `_UNUSED` `ifndef NDEBUG`. Doesn't nvim already have this?

Comment (lewis6991) [reply src/nvim/macros.h:233]: I couldn't find anything but I can have another look.

Comment (jamessan) [reply src/nvim/macros.h:233]: We just have code like:

```c
#ifndef NDEBUG
  assert(foo);
#else
  (void)foo;
#endif
```
or, when possible, putting the entire block of code inside the `NDEBUG` check.

File: src/nvim/macros.h [hunk 2]
```diff
@@ -226,4 +226,10 @@
 
 #define EMPTY_POS(a) ((a).lnum == 0 && (a).col == 0 && (a).coladd == 0)
 
+// Mark unused variables with _UNUSED, so that we can suppress warnings from
+// gcc's -Wunused-value and -Wunused-but-set-variable.
+// Prefix with _ to not clash with xdiff/xinclude.h which defines UNUSED for use
```
Comment (jamessan) [inline src/nvim/macros.h:231]: mpack is a separate package, which we just happen to vendor currently.  I don't think we need to add this to nvim's headers.

**Review by jamessan — APPROVED** [review]
LGTM, but I'll defer to @bfredl's opinion

**Comment by bfredl** [comment]
Hmm, normally we would just add a `(void)x` statement without any macro, though this file is not covered by our style guide so maybe it doesn't matter in this case.

**Comment by lewis6991** [comment]
> Hmm, normally we would just add a `(void)x` statement without any macro, though this file is not covered by our style guide so maybe it doesn't matter in this case.

Done

**Comment by bfredl** [comment]
thanks :)

**Pull request merged**